h/cli/commands/shell.py

```diff
@@ -1,83 +0,0 @@
-# pylint: disable=redefined-outer-name
-"""Command to run an interactive shell with application context."""
-
-import click
-
-BANNER = """Environment:
-  m, models    The `h.models` module.
-  registry     Active Pyramid registry.
-  request      Active request object.
-  session      Active database session.
-"""
-
-
-def autodetect():
-    try:
-        import bpython
-
-        return "bpython"
-    except ImportError:
-        try:
-            import IPython
-
-            return "ipython"
-        except ImportError:
-            pass
-
-    return "plain"
-
-
-def bpython(**locals_):
-    import bpython
-
-    bpython.embed(locals_, banner=BANNER)
-
-
-def ipython(**locals_):
-    # pylint: disable-all
-    from IPython import start_ipython
-    from traitlets.config import get_config
-
-    c = get_config()
-    c.TerminalInteractiveShell.banner2 = BANNER
-    start_ipython(argv=[], config=c, user_ns=locals_)
-
-
-def plain(**locals_):
-    import code
-
-    code.interact(banner=BANNER, local=locals_)
```

Comment:
This is never used unless you manually hack your tox environment to uninstall `ipython`